crates/sui-framework/packages/sui-system/tests/rewards_distribution_tests.move

```diff
@@ -620,197 +614,143 @@ fun uncapped_rewards() {
-        .inner_mut_for_testing()
-        .advance_epoch(
-            start_epoch + 1,
-            65,
-            balance::zero(),
-            balance::zero(),
-            0,
-            0,
-            0,
-            0,
-            epoch_start_time,
-            ctx,
-        )
-        .destroy_for_testing(); // balance returned from `advance_epoch`
-    ctx.increment_epoch_number();
+    let opts = runner.advance_epoch_opts().protocol_version(65).epoch_start_time(epoch_start_time);
+    runner.advance_epoch(option::some(opts)).destroy_for_testing();
 
     // should distribute 3 epochs worth of subsidies: 560, 561, 562
-    assert_eq!(sui_system.get_stake_subsidy_distribution_counter(), start_distribution_counter + 3);
-    check_distribution_counter_invariant(&mut sui_system, ctx);
+    runner.system_tx!(|system, ctx| {
+        assert_eq!(ctx.epoch(), 563);
+        assert_eq!(system.epoch(), 563);
+        assert_eq!(system.get_stake_subsidy_distribution_counter(), start_distribution_counter + 3);
+        check_distribution_counter_invariant(system, ctx);
+    });
 
     // ensure that next epoch change only distributes one epoch's worth
-    sui_system
-        .inner_mut_for_testing()
-        .advance_epoch(
-            start_epoch + 2,
-            65,
-            balance::zero(),
-            balance::zero(),
-            0,
-            0,
-            0,
-            0,
-            epoch_start_time + epoch_duration,
-            ctx,
-        )
-        .destroy_for_testing(); // balance returned from `advance_epoch`
-    ctx.increment_epoch_number();
+    let opts = runner
+        .advance_epoch_opts()
+        .protocol_version(65)
+        .epoch_start_time(epoch_start_time + epoch_duration);
+    runner.advance_epoch(option::some(opts)).destroy_for_testing();
 
     // should distribute 1 epoch's worth of subsidies: 563 only
-    assert_eq!(sui_system.get_stake_subsidy_distribution_counter(), start_distribution_counter + 4);
-    check_distribution_counter_invariant(&mut sui_system, ctx);
+    runner.system_tx!(|system, ctx| {
+        assert_eq!(ctx.epoch(), 564);
+        assert_eq!(system.epoch(), 564);
+        assert_eq!(system.get_stake_subsidy_distribution_counter(), start_distribution_counter + 4);
+        check_distribution_counter_invariant(system, ctx);
+    });
 
-    test_scenario::return_shared(sui_system);
-    test.end();
+    runner.finish();
 }
 
 #[test]
-// TODO: come back to me once safe mode emulation is implemented
-fun stake_subsidy_with_safe_mode_epoch_563_to_564() {
-    set_up_sui_system_state_with_big_amounts();
-
-    let mut test = test_scenario::begin(VALIDATOR_ADDR_1);
-    let mut sui_system = test.take_shared<SuiSystemState>();
-    let ctx = test.ctx();
-    // mimic state during epoch 563, if we're in safe mode since the 560 -> 561 epoch change
-    let start_epoch: u64 = 563;
-    let start_distribution_counter = 540;
+fun stake_subsidy_with_safe_mode_epoch_562_to_563_2() {
```

Comment:
I would keep the old name + the old comment at 701. Otherwise, it's hard to figure out what these tests are for. And maybe add a new comment about why we need both tests (this one is testing recovering from safe mode after 4 epochs, 3 for the previous one)